workflows/metatdenovo.nf

```diff
@@ -73,17 +144,373 @@ workflow METATDENOVO {
+        ch_gff          = UNPIGZ_GFF.out.unzipped
+        ch_protein      = PRODIGAL.out.faa
+        ch_versions     = ch_versions.mix(PRODIGAL.out.versions)
+    }
+
+    //
+    // SUBWORKFLOW: run TRANSDECODER. Orf caller alternative for eukaryotes.
+    //
+    if ( orf_caller == 'transdecoder' ) {
+        TRANSDECODER ( ch_assembly_contigs.map { meta, contigs -> [ [id: "transdecoder.${meta.id}" ], contigs ] } )
+        ch_gff      = TRANSDECODER.out.gff
+        ch_protein  = TRANSDECODER.out.pep
+        ch_versions = ch_versions.mix(TRANSDECODER.out.versions)
+    }
+
+    // Populate channels if the user provided the orfs
+    if ( orf_caller == 'user_orfs' ) {
+        Channel
+            .value ( [ [ id: "${assembler}.${orf_caller}" ], file(params.gff) ] )
+            .set { ch_gff }
+        Channel
+            .value ( [ [ id: "${assembler}.${orf_caller}" ], file(params.protein_fasta) ] )
+            .set { ch_protein }
+    }
+
+    //
+    // MODULE: Create a BBMap index
+    //
+    BBMAP_INDEX(ch_assembly_contigs.map { meta, contigs -> contigs })
+    ch_versions   = ch_versions.mix(BBMAP_INDEX.out.versions)
+
+    //
+    // MODULE: Call BBMap with the index once per sample
+    //
+    BBMAP_ALIGN ( ch_clean_reads, BBMAP_INDEX.out.index )
+    ch_versions = ch_versions.mix(BBMAP_ALIGN.out.versions)
+
+    //
+    // SUBWORKFLOW: classify ORFs with a set of hmm files
+    //
+    ch_hmmrs
+        .combine(ch_protein)
+        .map { hmm, meta, protein ->[ [ id: "${assembler}.${orf_caller}" ], hmm, protein ] }
+        .set { ch_hmmclassify }
+    HMMCLASSIFY ( ch_hmmclassify )
+    ch_versions = ch_versions.mix(HMMCLASSIFY.out.versions)
+
+    //
+    // MODULE: FeatureCounts. Create a table for each samples that provides raw counts as result of the alignment.
+    //
+
+    BAM_SORT_STATS_SAMTOOLS ( BBMAP_ALIGN.out.bam, ch_assembly_contigs )
+    ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions)
+
+    BAM_SORT_STATS_SAMTOOLS.out.bam
+        .combine(ch_gff.map { meta, bam -> bam } )
+        .set { ch_featurecounts }
+
+    ch_collect_stats
+        .combine(BAM_SORT_STATS_SAMTOOLS.out.idxstats.collect { meta, idxstats -> idxstats }.map { [ it ] } )
+        .set { ch_collect_stats }
+
+    FEATURECOUNTS_CDS ( ch_featurecounts)
+    ch_versions       = ch_versions.mix(FEATURECOUNTS_CDS.out.versions)
+
+    //
+    // MODULE: Collect featurecounts output counts in one table
+    //
+    FEATURECOUNTS_CDS.out.counts
+        .collect() { meta, featurecounts -> featurecounts }
+        .map { featurecounts -> [ [ id:"${assembler}.${orf_caller}" ], featurecounts ] }
+        .set { ch_collect_feature }
+
+    COLLECT_FEATURECOUNTS ( ch_collect_feature )
+    ch_versions           = ch_versions.mix(COLLECT_FEATURECOUNTS.out.versions)
+    ch_fcs_for_stats      = COLLECT_FEATURECOUNTS.out.counts.collect { meta, tsv -> tsv }.map { [ it ] }
```

Comment:
Again, I'm confused by the need of `.map { [ it ] }` where it is already a list.

Reply:
see above